libgadget/winds.c

```diff
@@ -471,18 +440,25 @@ sfr_wind_feedback_ngbiter(TreeWalkQueryWind * I,
     double p = windeff * I->Mass / I->TotalWeight;
     double random = get_random_number(I->base.ID + P[other].ID);
     if (random < p) {
-        /* in this case the particle is already locked by the tree walker */
-        /* we may want to add another lock to avoid this. */
-        if(P[other].ID != I->base.ID)
-            lock_spinlock(other, WIND_GET_PRIV(lv->tw)->spin);
+        double dir[3];
+        get_wind_dir(other, dir);
+        int j;
+        for(j = 0; j < 3; j++)
+        {
+            #pragma omp atomic update
+            P[other].Vel[j] += v * dir[j];
```

Comment:
I think this is different from the old behavior. the scope of the critical region has reduced -- you are using 3, not 1. 

Were we only locking the other particle here to make sure velocity change is atomic?

Reply:
In the isotropic winds case yes - the complex case was the non-isotropic wind, which uses the current velocity to work out the new wind direction. I tried to think of clever ways to get around it but since we don't use it and most people have moved away from it I thought: why bother?

Reply:
I thought the non-isotropic wind only uses (should only use) a reference direction computed from nearby DM particles or something like that, to compute the kick momentum, which is then added to the current velocity of P[other] just like this.

The kick momentum shouldn't depend on the current velocity of other, then?

Reply:
In principle I agree that the non-isotropic wind should do something like that, but what it actually does is that the wind direction is set by (v - vmean) . a   

vmean is set from the nearby DM particles and a is the short-range gravitational force. 

I thought about this and...I rather feel that this is essentially a random number: v may depend on whether the particle has already been kicked, and the short-range force is likely very much not the total acceleration. Since the physics seem at best confused, I thought maybe best to just remove it.